### PR TITLE
create hero component

### PR DESCRIPTION
**Pull request by wibjorn**
Task: task-659504

Link: preview-420

Pull request 420! Time to be a hero.

This pull request adds a hero and several pieces of documentation.

- Adds a hero component
-  Adds a "Full screen content section button" to the site, so that heroes can properly be tested
- Creates a layout gap placeholder class
- Updates parcel transformer to include an html code block that can be full bleed when content section is full screens
- Adds page for background-image pattern atomics

## Testing

https://design.docs.microsoft.com/pulls/420/components/hero.html
https://design.docs.microsoft.com/pulls/420/patterns/hero.html
https://design.docs.microsoft.com/pulls/420/atomics/background-pattern.html

1. Step one
2. Step two
   Expected result:

## Additional information

[Optional]

### CODE REVIEW

File: site/src/components/hero.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		<h2 class="font-size-h2">A hero with a primary background color and inverted text</h1>
+	</div>
+</section>
+```
+
+That is not quite where the story ends. You can also use certain classes on the `.hero` element to specify a background pattern.
+
+### Hero sizes
+
+Heroes have defaults spacing that relies on the `.hero-content` element, so be sure to include it. However, there are also a few modifiers that can be used to change the _minimum_ height of the hero. If you pack a hero with text, it will grow in height. Each size is denoted by a t-shirt size. For more on how Atlas creates its names [check out the explainer for component sizes](../components/overview.md).
```
Comment (homing1) [inline site/src/components/hero.md:48]: super nit: default

File: site/src/components/hero.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+<section class="hero hero-lg background-color-tertiary color-tertiary-invert">
+	<div class="hero-content">
+		<h2 class="font-size-h2">Large</h2>
+	</div>
+</section>
+```
+
+### Hero images
+
+Heroes have two primary ways of displaying media. The first is with the `.hero-image` class, which should be place adjacent to the `.hero` class. This subclass can be used in conjunction with inline css variable declarations to display images suitable for different themes. Image sources should be reference in a style attribute as shown below. This allows for convenient image swapping between themes. These images display to the side of the hero's content. On narrow screens, no image will display.
```
Comment (homing1) [inline site/src/components/hero.md:84]: super nit: placed

File: site/src/components/hero.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+			--hero-background-image-dark: url('https://via.placeholder.com/1000x500');">
+	<div class="hero-content">
+		<h2 class="font-size-h2">Hero</h1>
+	</div>
+</section>
+```
+
+## Heroes with full media backgrounds
+
+Hero backgrounds with full-bleed media can be somewhat problematic. Always be sure that you can ensure accessible text colors between a background image and a piece of text. You may also need to apply a static theme class to your hero to ensure the text does not change with the theme. Images and videos included in this way will scale always cover the entire contains, so responsive testing is particularly important when validating media choices.
```
Comment (homing1) [inline site/src/components/hero.md:101]: supernit: contains -> container?

File: site/src/patterns/hero.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+---
+title: Hero
+description: Common hero patterns in the Atlas Design System
+template: standard
+---
+
+# Hero Patterns
+
+This page catelogues some of the common combinations that might but used with the hero component. For component documentation, [visit the hero component page](../components/hero.md).
```
Comment (homing1) [inline site/src/patterns/hero.md:9]: supernit: catalogues

File: site/src/components/hero.md (hunk 2 above)
Comment (wibjorn) [reply site/src/components/hero.md]: Updated the four typos you caught, thank you!

File: css/src/components/hero.scss [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+		padding-block: $hero-padding-sm;
+
+		@include desktop {
+			min-height: $hero-min-height-default;
+			padding-block: $hero-padding-default;
+			padding-inline-end: $hero-padding-sm;
+		}
+	}
+
+	&.hero-xs {
```
Comment (homing1) [inline css/src/components/hero.scss:33]: missing .hero-content here

image

Comment (wibjorn) [reply css/src/components/hero.scss:33]: updated

**Comment by wibjorn** [comment]
@ollips Is taking over this PR (or making a new one)

File: css/src/atomics/index.scss [hunk 6]
```diff
@@ -1,5 +1,6 @@
 @import './typography.scss';
 @import './border.scss';
+@import './background-pattern.scss';
```
Comment (wibjorn) [inline css/src/atomics/index.scss:3]: shoot, file should be background-image?

Comment (wibjorn) [reply css/src/atomics/index.scss:3]: don't feel strongly about it though

Comment (ollips) [reply css/src/atomics/index.scss:3]: New PR for comment is here - https://github.com/microsoft/atlas-design/pull/427 :) Closing the current one

**Pull request closed without merging**